File: adapters/frontend/swift/Sources/Rampart/RampartClaims.swift
import Foundation

/// User profile claims returned by the Rampart `/me` endpoint.
public struct RampartClaims: Codable, Sendable, Equatable {

    /// User ID (UUID), corresponds to the JWT "sub" claim.
    public let sub: String

    /// Organization ID (UUID).
    public let orgId: String

    /// Preferred username.
    public let preferredUsername: String?

    /// Email address.
    public let email: String

    /// Whether the email has been verified.
    public let emailVerified: Bool

    /// Roles assigned to the user.
    public let roles: [String]

    /// First name.
    public let givenName: String?

    /// Last name.
    public let familyName: String?

    /// Whether the account is active.
    public let enabled: Bool?

    /// Account creation timestamp (ISO 8601).
    public let createdAt: String?

    /// Last update timestamp (ISO 8601).
    public let updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case sub = "id"
        case orgId = "org_id"
        case preferredUsername = "preferred_username"
        case email
        case emailVerified = "email_verified"
        case roles
        case givenName = "given_name"
        case familyName = "family_name"
        case enabled
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }

    public init(
        sub: String,
        orgId: String,
        preferredUsername: String? = nil,
        email: String,
        emailVerified: Bool = false,
        roles: [String] = [],
        givenName: String? = nil,
        familyName: String? = nil,
        enabled: Bool? = nil,
        createdAt: String? = nil,
        updatedAt: String? = nil
    ) {
        self.sub = sub
        self.orgId = orgId
        self.preferredUsername = preferredUsername
        self.email = email
        self.emailVerified = emailVerified
        self.roles = roles
        self.givenName = givenName
        self.familyName = familyName
        self.enabled = enabled
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

File: adapters/frontend/kotlin/src/main/kotlin/com/rampart/RampartClaims.kt
package com.rampart

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * User profile claims returned by the Rampart `/me` endpoint.
 */
@Serializable
data class RampartClaims(
    /** User ID (UUID), corresponds to the JWT "sub" claim. */
    @SerialName("id") val sub: String,

    /** Organization ID (UUID). */
    @SerialName("org_id") val orgId: String,

    /** Preferred username. */
    @SerialName("preferred_username") val preferredUsername: String? = null,

    /** Email address. */
    @SerialName("email") val email: String,

    /** Whether the email has been verified. */
    @SerialName("email_verified") val emailVerified: Boolean = false,

    /** Roles assigned to the user. */
    @SerialName("roles") val roles: List<String> = emptyList(),

    /** First name. */
    @SerialName("given_name") val givenName: String? = null,

    /** Last name. */
    @SerialName("family_name") val familyName: String? = null,

    /** Whether the account is active. */
    @SerialName("enabled") val enabled: Boolean? = null,

    /** Account creation timestamp (ISO 8601). */
    @SerialName("created_at") val createdAt: String? = null,

    /** Last update timestamp (ISO 8601). */
    @SerialName("updated_at") val updatedAt: String? = null,
)
